src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction-analysis.component.spec.ts

```diff
@@ -95,10 +88,11 @@ describe('ProgrammingExerciseInstructionInstructorAnalysis', () => {
             fixture.detectChanges();
 
             // Check internal state and output of component.
-            expect(comp.missingTestCases).to.deep.equal(missingTestCases);
-            expect(comp.invalidTestCases).to.deep.equal(invalidTestCases);
-            expect(comp.invalidHints).to.deep.equal(invalidHints);
-            expect(emitAnalysisSpy).to.have.been.calledOnceWithExactly(completeAnalysis);
+            expect(comp.missingTestCases).toEqual(missingTestCases);
+            expect(comp.invalidTestCases).toEqual(invalidTestCases);
+            expect(comp.invalidHints).toEqual(invalidHints);
+            expect(emitAnalysisSpy).toHaveBeenCalledWith(completeAnalysis);
+            expect(emitAnalysisSpy).toHaveBeenCalledTimes(1);
```

Comment:
I think first checking the count and then the parameters is a bit better since the parameters are more specific

Reply:
:+1: changed in 4387660066771df467461d7d8492c0da978f8ff0